osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -106,16 +115,21 @@ public virtual void Seek(double time)
         /// <summary>
         /// Resets this <see cref="GameplayClockContainer"/> and the source to an initial state ready for gameplay.
         /// </summary>
-        public virtual void Reset()
+        /// <param name="startClock">Whether to start the clock immediately, if not already started.</param>
+        /// <param name="gameplayStartTime">A time to use for future <see cref="Reset"/> calls as the definite start of gameplay.</param>
+        public void Reset(bool startClock = false, double? gameplayStartTime = null)
         {
-            ensureSourceClockSet();
-            Seek(0);
+            if (gameplayStartTime != null)
+                StartTime = gameplayStartTime;
```

Comment:
Not sure how to feel about this param and it being used like this, it's a weird edgecasey state mutation in a method that is supposed to be a "reset" method. Is it an option to remove it from `Reset()` and instead explicitly set `StartTime` in call sites that need it?

Reply:
I've made this change, I think it feels better.

Reply:
Is this PR missing a commit? The parameter's still here.

Reply:
You might be looking at an old diff, it's definitely not there for me